### ISSUE TITLE
Choose what section should notifications open when tapped

### ISSUE THREAD

**Issue by Manuito83** [Manuito83/torn-pda#111]

**Comment by alxspiker** [comment]
### Suggestion: Implement a "Notification Taps" Setting

A good way to tackle this would be to introduce a new section in the main settings page, perhaps called **"Notification Taps"** or **"Notification Behavior"**.

This new section could contain a list of all the different notification types a user can receive (e.g., Travel, Life, Energy, Messages, etc.). Next to each type, a dropdown menu would allow the user to select the desired action when that notification is tapped.

The options in the dropdown could include:
*   **Open in Browser (Default)**
*   **Open 'Profile' Section**
*   **Open 'Travel' Section**
*   **Open 'Items' Section**
*   And so on for all relevant sections in the app.

This approach would empower users to tailor the app to their specific workflows.

### Proposed Implementation

To make this happen, we could make the following adjustments to the codebase:

**1. Update `SettingsProvider` to manage preferences:**

First, we'd need to expand `lib/providers/settings_provider.dart` to store and retrieve these new settings.

```dart
// In lib/providers/settings_provider.dart

// A map to store the notification tap actions for each type
Map<String, String> _notificationTapActions = {};

// A getter for the notification tap actions
Map<String, String> get notificationTapActions => _notificationTapActions;

// Method to update the tap action for a given notification type
Future<void> setNotificationTapAction(String notificationType, String section) async {
  _notificationTapActions[notificationType] = section;
  // This will also save the preference to the device's storage
  await Prefs().setNotificationTapAction(notificationType, section);
  notifyListeners();
}

// In the loadPreferences method, we would add:
_notificationTapActions = await Prefs().getNotificationTapActions();
```

**2. Enhance the Settings Page UI:**

In `lib/pages/settings_page.dart`, a new UI section would be created. This would likely involve adding a `ListView` of `ListTile` widgets, each with a `DropdownButton` to allow the user to select the desired action for each notification type.

**3. Modify Notification Handling Logic:**

The core of the change would be in `lib/drawer.dart`. The `_onFirebaseBackgroundNotification` and `_onForegroundNotification` methods would be updated to consult the `SettingsProvider` before acting.

For example, the current logic for a "travel" notification is hardcoded:

```dart
// Current logic in lib/drawer.dart
if (travel) {
  launchBrowserWithUrl = true;
  browserUrl = "https://www.torn.com";
}
```

This would be replaced with a more flexible approach that respects the user's choice:

```dart
// Suggested new logic in lib/drawer.dart
if (travel) {
  // Get the user's preference, defaulting to 'Browser' if not set
  final section = _settingsProvider.notificationTapActions['travel'] ?? 'Browser';

  if (section == 'Browser') {
    launchBrowserWithUrl = true;
    browserUrl = "https://www.torn.com";
  } else {
    // Find the index of the desired section and open it
    final sectionIndex = _drawerItemsList.indexOf(section);
    if (sectionIndex != -1) {
      _callSectionFromOutside(sectionIndex);
    }
  }
}
```

This same pattern would be applied to all other relevant notification types (`hospital`, `life`, `energy`, etc.), creating a robust and highly customizable notification system.

**Comment by AlexTzib** [comment]
Hi @Manuito83, I took a look at this one before starting work and it seems part of the original request is already covered now.

From the current code, Torn PDA already has configurable tap actions for:
- Travel notifications
- Travel Live Activity / Live Update taps
- Life notifications
- Drugs cooldown notifications
- Medical cooldown notifications
- Booster cooldown notifications

Before opening a PR here, I think it would be better to confirm the missing scope instead of building a broad generic settings system. Which notification types would you still like to make configurable? For example: energy, nerve, hospital, jail, racing, messages/events, forums, retals/assists, etc.

Once the remaining list is clear, we can keep the PR focused and avoid changing behavior that is already working.
